chapters/analysis.typ

```diff
@@ -28,45 +21,46 @@ In @ast_analysis, abstract syntax trees are looked at, including their role in c
+In a lot of cases, automated tests are used to do so.
+Even if all tests succeed, one would theoretically still be required to prove that the expected result has the same behavior as the test input,
 however, due to these tests being very concise, their correctness can typically be verified through a quick inspection.
-The testing of refactorings is explored in more detail in @testing.
-
-// COR Different refactorings vary in complexity.
-// For example renaming of a local variable, typically, has limiited potential for unexpected side-effects.
-// In most cases it is sufficient to check whether the new name already exists in the affected scope.
-//
-// Our "Abbreviate Function Template" refactoring, on the other hand, could have surprising side-effects, in which cases it must not be applied.
-// (Beispiel könnte noch um eine Variante ergänzt werden, welcher die Template-Argumente deduziert werden - Normalfall).
-//
-// ... Zudem sollte man auch noch beachten, dass in einigen Fällen nicht der ganze Code analyisert werden kann, bzw. gar nicht verfügar ist.
-//
-// Zusätzliche Schwierigkeit in C++: Präprozessor, womit quasi fast jeder Name umdefiniert werden kann. 
-In @refactoring_bad_example an example of a bad refactoring is shown.
+The testing of refactorings in clangd is explored in more detail in @testing. 
+
+Different refactorings vary in complexity.
+For example, renaming a local variable typically has limited potential for unexpected side-effects.
+In most cases, it is sufficient to check whether the new name already exists in the affected scope.
+The "Abbreviate Function Template" refactoring (@abbreviate_function_template), on the other hand, could have surprising side-effects,
+in which case it must not be applied.
+It also should be considered that in some cases, not the whole code can be analyzed or is not even available.
+
+// COR Zusätzliche Schwierigkeit in C++: Präprozessor, womit quasi fast jeder Name umdefiniert werden kann. 
+// VINA No idea how to document this...
+In @refactoring_bad_example an example of bad refactoring is shown.
 A function is defined with the template type parameters `T` and `U` and the function parameters `p1` and `p2`, which use the template type parameters in reverse order.
-// COR It is not a refactoring then, just a code transformation/change.
-If a refactoring, for example one which converts the functions to their abbreviated form using `auto` parameters, would blindly use `auto` for all function parameter types, it would result in a different function signature.
-The compiler will then throw an error at the call-site, because the function call is no longer valid.
+If a refactoring, for example, converts the functions to their abbreviated form using `auto` parameters and blindly uses `auto` for all function parameter types, it would result in a different function signature.
+The compiler will then throw an error at the call-site because the function call is no longer valid.
```

Comment:
```suggestion
The compiler may throw an error at the call-site because the function call is no longer valid, but the updated source code could also just compile if the parameter types are all inferred.
```